#!/usr/bin/env python
PACKAGE = "opencv3mixing"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("high_thresh", double_t, 0, "The first of threshold", 50.0, 0, 255)
gen.add("low_thresh", double_t, 0, "The second of threshold", 200.0, 0, 255)
size_enum = gen.enum([gen.const("1", int_t, 1, ""),
                      gen.const("3", int_t, 3, ""),
                      gen.const("5", int_t, 5, ""),
                      gen.const("7", int_t, 7, "")],
                      "An enum for apertureSize")
gen.add("apertureSize", int_t, 0, "The apertureSize for Sobel", 3, 1, 7, edit_method=size_enum)
gen.add("L2gradient", bool_t, 0, "L2gradient = true is better & heavy")

exit(gen.generate(PACKAGE, "edge_detector", "EdgeParam"))